persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -371,16 +371,31 @@ public MarshalledEntry load(Object key) {
     @Override
     public void writeBatch(Iterable<MarshalledEntry<? extends K, ? extends V>> marshalledEntries) {
         try {
+            // Store a list of WriteBatch objects so that we can perform all marshalling before acquiring the semaphore
+            List<WriteBatch> writeBatches = new ArrayList<>();
+            int batchSize = 0;
             WriteBatch writeBatch = new WriteBatch();
-            for (MarshalledEntry entry : marshalledEntries)
+            for (MarshalledEntry entry : marshalledEntries) {
```

Comment:
Hrmm I don't really like the way batching is done here. We end up putting _all_ entries into batches, but retain all batches. We should really write each batch as it is filled as we do with the other stores and then clear it.